.azure-pipelines/templates/build.yml

```diff
@@ -0,0 +1,87 @@
+parameters:
+- name: py_version
+  type: string
+  default: ''
+- name: image
+  type: string
+  default: ''
+- name: ccache
+  type: boolean
+  default: false
+- name: OSX_VERSION
+  type: string
+  default: ''
+- name: publish
+  type: string
+  default: ''
+
+jobs:
+  - job:
+    timeoutInMinutes: 120
+    displayName: ${{ format('{0}-py{1}', parameters.image, parameters.py_version) }}
+    pool:
+      vmImage: ${{ parameters.image }}
+    variables:
+      - name: OSX_VERSION
+        value: ${{ parameters.OSX_VERSION }}
+
+    steps:
+      - checkout: self
+        submodules: true
+
+      - ${{ if eq(parameters.ccache, true) }}:
+        - template: ccache.yml
+
+      # Note that using set -ex when setting vso variables messes up paths
+      - ${{ if contains(parameters.image, 'windows') }}:
+        - bash: |
+            echo "##vso[task.prependpath]$CONDA\Scripts"
+            echo "##vso[task.setvariable variable=file_extension]bat"
+          displayName: 'Configure Conda path'
+      - ${{ if not(contains(parameters.image, 'windows')) }}:
+        - bash: |
+            echo "##vso[task.prependpath]$CONDA/bin"
+            echo "##vso[task.setvariable variable=file_extension]sh"
+          displayName: 'Configure Conda path'
+      - bash: echo "##vso[task.setvariable variable=conda_dir]$CONDA"
+        displayName: 'Configure Conda dir'
+      - ${{ if not(eq(length(parameters.OSX_VERSION), 0)) }}:
+        - bash: |
+            set -ex
+            sudo chown -R $USER $CONDA
+          displayName: 'Take ownership of Conda installation'
+
+      - bash: |
+          set -ex
+          conda --version
+          conda install --yes conda-build
+          conda config --set always_yes yes --set changeps1 no
+          # Specify python version in env file
+          sed 's/- python$/- python=${{ parameters.py_version }}/g' scipp-developer-minimal.yml > temp-env.yml
```

Comment:
Minor point: Env name will not match actual file that produced it, maybe that's confusing if someone then wonders why `scipp-developer-minimal` is using python 3.8 say, but that is not specified for `scipp-developer-minimal.yml`?

Reply:
This only happens temporarily on azure, i think it's fine to have it like that. I think the benefits of not needing a new copy of the env file for each python version outweigh the small inconsistency you point out.